### ISSUE TITLE
Add a global parameter filter to TM_Design, with a first implementation of MaxSize

### ISSUE THREAD

**Issue by DinisCruz** [TeamMentor/TM_4_0_Design#370]
The idea is to apply a global max size to all user provided data:

* url parameters
* post post parameters
* cookies
* headers

This filter could have a number of side effects and will need to have a way to be bypassed by the code (to handle exceptions)

There are lot of synergies between this filter and https://github.com/TeamMentor/TM_4_0_Design/issues/369 (although the filter could be done first)

Related to https://github.com/TeamMentor/TM_4_0_Design/issues/325

**Comment by roman87** [comment]
I am not sure why we are having this global maxSize. Each parameter needs to be able to validate its input not only based on size but based on what it actually expects. This includes the size. The size for a filter (keyword/query) parameter is quite different than for a URL than for a search string. So lets discuss, when the time comes. For now this is a P2 in sprint 3. (we are now in sprint 1/2)

**Comment by DinisCruz** [comment]
exactly, this is not an easy problem to solve, and my view is that there are very few cases when this type of data validation/sanitisation should be a TM Design responsibility.

There is a case for max length, and that can be enforced globally

Specific encoding or validation, will need to be done by each controller (or final rest api)
